gw_eccentricity/load_data.py

```diff
@@ -1030,6 +1275,100 @@ def reomve_junk_from_nr_data(t, modes_dict, num_orbits_to_remove_as_junk):
+
+    Returns
+    -------
+    t_clean: array-like
+        Time array corresponding to clean NR data.
+
+    modes_dict_clean: dict
+        modes_dict with `num_orbits_to_remove_as_junk` orbits removed from the
+        begining of modes array.
+    """
+    num_orbits_duration = get_num_orbits_duration_from_horizon_data(
+        horizon_filepath, num_orbits_to_remove_as_junk)
+    idx_junk = np.argmin(np.abs(t - (t[0] + num_orbits_duration)))
+    t_clean = t[idx_junk:]
+    modes_dict_clean = {}
+    for key in modes_dict:
+        modes_dict_clean[key] = modes_dict[key][idx_junk:]
+
+    return t_clean, modes_dict_clean
+
+
+def get_num_orbits_duration_from_horizon_data(horizon_filepath, num_orbits,
+                                              deltaTOverM=1):
+    """Get the duration of `num_orbits` from phase data in `Horizons.h5`.
+
+    Parameters
+    ----------
+    horizon_filepath: str
+        Path to the `Horizons.h5` file. This file typically can be found in the
+        same directory where the waveform files are located.
+
+    num_orbits: float
+        Number orbits from the start of the simulation. This function estimates
+        the duration of `num_orbits` from orbital phase which is obtained from
+        the coordinates of the black holes stored in the `Horizons.h5` file.
+
+    deltaTOverM: float, default=1
+        Time step in dimensionless units. It is used to interpolate the data
+        since the NR data in `Horizons.h5` are not stored at uniform time grid.
+
+    Returns
+    -------
+    num_obits_duration : float
+        Duration of first `num_orbits`.
+    """
+    horizons_data = h5py.File(horizon_filepath, "r")
+    xA_data = horizons_data["AhA.dir"]["CoordCenterInertial.dat"]
+    xB_data = horizons_data["AhB.dir"]["CoordCenterInertial.dat"]
+    time = xA_data[:, 0]
+    separion_vec = xA_data[:, 1:] - xB_data[:, 1:]
+    # We will assume the system to be non-precessing so that the motion
+    # is restricted in the x-y plane and the z-coordinate is effectively zero.
+    # Check if it is true, i.e., z-coordinate remains very small
+    if any(np.abs(separion_vec[:, 2]) > 1e-5):
+        raise Exception(
+            "The system seems to be precessing. Cannot get the orbital "
+            "phase by assuming the motion is restricted in the x-y plane.")
+    # Get the orbital phase
+    phase_orb = np.unwrap(np.arctan2(separion_vec[:, 1], separion_vec[:, 0]))
+    # Interpolate phase
+    time_interp = np.arange(min(time), max(time), deltaTOverM)
+    phase_interp = InterpolatedUnivariateSpline(time, phase_orb)(time_interp)
+    # Find the duration of first num_orbits assuming that the orbital phase
+    # changes by 2pi over one orbit
+    idx_at_num_obits_from_start = np.argmin(
+        np.abs(phase_interp - (phase_interp[0] + num_orbits * 2 * np.pi)))
+    num_obits_duration = (time_interp[idx_at_num_obits_from_start]
+                          - time_interp[0])
+    return num_obits_duration
```

Comment:
Have you tested that this solves the problem you are trying to solve, let's say for the Ecc3dSur dataset?

Reply:
I have not tried the full set yet. I just tried the one I was having a problem with. Here is one plot. I clearly show that Horizons data gives a much better result
![image](https://github.com/vijayvarma392/gw_eccentricity/assets/44096766/bc9ab6ff-02b7-400f-9d00-b65fd9ebac8b)

Reply:
Is that for two orbits? Visually, it still looks like less than two. 
I might be because:
The time array in Horizons.h5 is the time coordinate of the simulation, so it starts at t=0.
The time array in the waveform file is related to the retarded, shifted back to the origin, which means it starts at some t<0, like order -100M or so. However, the t=0 point in both of these files can be approximately associated with each other.

What if you do: get DeltaT from Horizons.h5, then truncate all times up to t_waveform = DeltaT, not t_waveform = t_waveform[0] + DeltaT. Does that make it better?

Reply:
Okay, I will check this.

Reply:
I shared some plots here. The first thing we see is that the duration of the waveform is random as it is obtained using noisy data. Comparing the duration from Horizons.h5, from $t=0$, using just $t >$ duration seems to give a better result than using $t > t[0]$ + duration. What do you think?

![image](https://github.com/vijayvarma392/gw_eccentricity/assets/44096766/dd959ca1-ff8d-454a-80a0-4550560b7c85)
![image](https://github.com/vijayvarma392/gw_eccentricity/assets/44096766/b88a612a-8dbf-440b-b12c-3a151c65f85b)
![image](https://github.com/vijayvarma392/gw_eccentricity/assets/44096766/6765ae76-db72-4849-9219-3d8e358582f9)
![image](https://github.com/vijayvarma392/gw_eccentricity/assets/44096766/1f6aa174-645e-4268-a058-24453ca3dcd3)

Reply:
Yeah, this is for the reason I mentioned above. Let's go with the pink line. But, you should test this more to be safe, like for the EccHybTest dataset.

Reply:
The pink line looks good for the `EccHybTest` cases also
![image](https://github.com/vijayvarma392/gw_eccentricity/assets/44096766/fffed10d-06b0-43b2-872a-2c66b3792969)
![image](https://github.com/vijayvarma392/gw_eccentricity/assets/44096766/5dca3576-35a7-4b2d-a342-a17846986b34)
![image](https://github.com/vijayvarma392/gw_eccentricity/assets/44096766/0016fed4-1ae0-4583-be53-69f827c8fb1a)

Reply:
sounds good